April 2025 refactor

PhilipMusk opened the pull request.

PhilipMusk committed:
Fix #3 Add Participant node - update properties

Within the Participant node:
Added case versus control indicator property

Added cancer_diagnosis_primary_site property inclusive of a temporary controlled vocabulary

Renamed the existing cancer_diagnosis property to cancer_diagnosis_morphology and added a temporary controlled vocabulary

Changed property type for Race from simple Enum to List plus Enum

Commented out body_mass_index property

Tidied up the Participant node to some degree

Elsewhere:
Added a binary data_collection_category_assessed property

PhilipMusk committed:
Fix #3 Add Participant node and push v1 of the yaml-based config file

Changed the name of the Participant cancer_diagnosis_morphology property to cancer_diagnosis_disease_morphology

Committed Version 1 of the yaml-based "soft" configuration file that will dictate the visual organization of the Data Collected tab

breadsp commented:
was able to incorporate the new files into the mock data generator and data loader, no issues

PhilipMusk committed:
Fix #3 Add Participant and yaml-based config file

Corrected two typos noted by Yizhen's review via Copilot

majensen committed:
@majensen fixes for MDF validation

- comment out "helper" properties ('prop_1',...)
- update dummy node name in submodel - data_collection ->
  data_collection_category, to corresponding with the
  dummy relationship 'associated_with'
- comment out 'person_email_address' property in model file, to
  correspond with commented-out property definition in props file.

majensen reviewed (approved):
This is all good with the updated I made in [5d6eee9](https://github.com/CBIIT/popsci-model/pull/5/commits/5d6eee9366d7ca16e116cf74b4807fdc5676ad9e) - now the popsci-submodel.yml also validates.

PhilipMusk commented:
Merging this PR per multiple reviews, and with all three model definition files in question having passed validation via GitHub Actions.
